src/query/graphite/native/aggregation_functions.go

```diff
@@ -278,6 +281,147 @@ func combineSeriesWithWildcards(
 	return r, nil
 }
 
+// splits a slice into chunks
+func chunkArrayHelper(slice []string, numChunks int) [][]string {
+	var divided [][]string
```

Comment:
nit: try to init with a size to avoid having go grow it automatically, since it does a silly doubling thing which can lead to a lot of allocs (each time capacity increases it allocs a new slice, copies, then frees the old one): https://play.golang.org/p/HaEISY70Ar8

 e.g. `divided := make([][]string, 0, numChunks)`

Reply:
Done!